perses/annihilation/relative.py

```diff
@@ -2541,3 +2537,269 @@ def _handle_exceptions(self):
+        core_bond_expression = f"{scale_bool_string} * (K/2)*(r-length)^2;"
+        bool_string = RxnHybridTopologyFactory.render_bool_string([''] + [f"lambda_{i}_bonds" for i in range(self._num_alchemical_regions)],
+                                                                    ['environment_region'] + [f"alchemical_region_{i}" for i in range(self._num_alchemical_regions)])
+
+        core_bond_expression += f"K = (1 - {bool_string}) * K1 + {bool_string} * K2;"
+        core_bond_expression += f"length = (1 - {bool_string})*length1 + {bool_string}*length2;"
+
+        custom_bond_force = openmm.CustomBondForce(core_bond_expression)
+
+        #add global parameters
+        for i in range(self._num_alchemical_regions):
+            custom_bond_force.addGlobalParameter(f"lambda_{i}_bonds", 0.0)
+
+        for i in ['scale_lambda', 'interscale_lambda']:
+            custom_bond_force.addGlobalParameter(i, 1.0)
+
+        #add per-bond parameter
+        for i in ['scale_region', 'nonscale_region', 'interscale_region']:
+            custom_bond_force.addPerBondParameter(i)
+
+        for i in ['environment_region'] + [f"alchemical_region_{i}" for i in range(self._num_alchemical_regions)]:
+            custom_bond_force.addPerBondParameter(i)
+
+        custom_bond_force.addPerBondParameter('length1') # old bond length
+        custom_bond_force.addPerBondParameter('K1') # old spring constant
+        custom_bond_force.addPerBondParameter('length2') # new bond length
+        custom_bond_force.addPerBondParameter('K2') # new spring constant
+
+        #now add the parameters
+        #there can only be a _single_ term for each atom pair, right?
+        
+        old_system_bond_force = self._old_system_forces['HarmonicBondForce']
+        new_system_bond_force = self._new_system_forces['HarmonicBondForce']
+
+        #make a list of hybrid-indexed bond terms
+        passed_old_terms, passed_new_terms = [], []
+        old_term_collector = {}
+        new_term_collector = {}
+
+        #gather the old system bond force terms into a dict
+        for term_idx in old_system_bond_force.getNumBonds():
+            p1, p2, r0, k = old_system_bond_force.getBondParameters(term_idx) #grab the parameters
+            hybrid_p1, hybrid_p2 = self._old_to_hybrid_map[p1], self._old_to_hybrid_map[p2] #make hybrid indices
+            sorted_list = sorted([hybrid_p1, hybrid_p2]) #sort the indices
+            assert not sorted_list in passed_old_terms, f"this bond already exists"
+            old_term_collector[sorted_list] = [r0, k]
+
+        # repeat for the new system bond force
+        for term_idx in new_system_bond_force.getNumBonds():
+            p1, p2, r0, k = new_system_bond_force.getBondParameters(term_idx)
+            hybrid_p1, hybrid_p2 = self._new_to_hybrid_map[p1], self._new_to_hybrid_map[p2]
+            sorted_list = sorted([hybrid_p1, hybrid_p2])
+            assert not sorted_list in passed_new_terms, f"this bond already exists"
```

Comment:
Nothing is being added to `passed_new_terms`, so this assert statement isn't meaningful